src/l3/ip.rs

```diff
@@ -107,6 +90,40 @@ pub union in6_u {
     pub u6_addr32: [u32; 4usize],
 }
 
+/// Protocol which is encapsulated in the IPv4 packet.
```

Comment:
seems to me strange or even incorrect to have to use `Ipv4Proto` for values that are not IPv4 specific - for example, `Tcp`, `Udp`, or even IPv6-specific such as `IPv6ICMP`.  The referenced Wikipedia page says:
> a list of the IP protocol numbers found in the field Protocol of the IPv4 header and the Next Header field of the IPv6 header

Could `Ipv4Proto` be changed to some name that would include both IPv4 and IPv6 - maybe `IPProto`?

Reply:
Yes, I will rename it to `IpProto`.